src/components/Field.js

```diff
@@ -7,10 +7,70 @@ class Field extends React.Component {
         }
     }
 
+
+    fenNotation = () => {
+        
+        var positions = [];
+     
+        this.props.fields.map((field) => (
+            positions.push(field.piece)
+        ))
+        
+        var rows = [];
+        var counter = 0;
+
+        for(var i = 0; i < 8; i++){
+            rows[i] = [];
+
+            for(var j = 0; j < 8; j++){
+                rows[i][j] = positions[counter];
+                counter++;
+            }
+            
+        }
+
+        var allshortArrays = '';
+
+        rows.forEach( (row, index) => {
+            
+            const shortArray = []; 
+            var emptyCounter = 1; 
+
+            for(let l = 0; l < 8; l++){
+
+                if(row[l] !== ''){
+            
+                    shortArray.push(row[l]);
+            
+                } else {
+                
+                    if(row[l + 1] ===''){
+                        emptyCounter++; 

```

Comment:
There is error here, let's say input array is 
`
            { id: 0, x: '1', y: '1', piece: 'R' },
            { id: 1, x: '2', y: '1', piece: '' },
            { id: 2, x: '3', y: '1', piece: 'B' },
            { id: 3, x: '4', y: '1', piece: '' },
            { id: 4, x: '5', y: '1', piece: '' },
            { id: 5, x: '6', y: '1', piece: '' },
            { id: 6, x: '7', y: '1', piece: 'N' },
            { id: 7, x: '8', y: '1', piece: 'R' },
`
Then result of this will be RN3B3R and it should be RN3B1R, you need to reset `RN3B1R` value when you are in previous loop.